db/migrate/20190422185437_add_location_table.rb

```diff
@@ -0,0 +1,27 @@
+class AddLocationTable < ActiveRecord::Migration[5.1]
+  def up
+    create_table :locations, force: true do |t|
+      t.string :name, default: "", null: false, comment: "Full display name, such as a complete address"
+      t.string :level_name, default: "", null: false, limit: 20, comment: "Level of specificity (country, state, subdivision, or city)"
+      t.string :country_name, default: "", null: false, limit: 100, comment: "Country (or equivalent) of this location if available"
+      t.string :state_name, default: "", null: false, limit: 100, comment: "State (or equivalent) of this location if available"
+      t.string :subdivision_name, default: "", null: false, limit: 100, comment: "Second-level administrative division (e.g. county/district/division/province/etc.) of this location if available"
+      t.string :city_name, default: "", null: false, limit: 100, comment: "City (or equivalent) of this location if available"
+
+      # Recommended settings for lat/lon
+      t.decimal :lat, precision: 10, scale: 6, comment: "The latitude of this location if available"
```

Comment:
I learned from https://gis.stackexchange.com/questions/8650/measuring-accuracy-of-latitude-and-longitude that 6 decimal places is accuracy within 11.1cm. So I guess this is sufficient! :)

Reply:
Do we have any desire to use mysql geo functions? https://dev.mysql.com/doc/refman/5.7/en/spatial-analysis-functions.html . Or use a gem like https://github.com/geokit/geokit-rails  ? If yes, we should optimize our schema for their designs.

Reply:
@gregdingle I think we'd like to leave the door open to geo functions in MySQL or Elasticsearch so I'll take a look at those schemas too. Our coordinate operations would probably look like "is this point in the polygon defined by these points" (with polygons in Elasticsearch or MySQL JSON if we upgrade to 5.7) but I'm not planning coordinate arithmetic beyond that yet